lib/gnat/command.ex

```diff
@@ -1,11 +1,26 @@
 defmodule Gnat.Command do
   @newline "\r\n"
+  @hpub "HPUB"
   @pub "PUB"
   @sub "SUB"
   @unsub "UNSUB"
 
   def build(:pub, topic, payload, []), do: [@pub, " ", topic, " #{IO.iodata_length(payload)}", @newline, payload, @newline]
   def build(:pub, topic, payload, [reply_to: reply]), do: [@pub, " ", topic, " ", reply, " #{IO.iodata_length(payload)}", @newline, payload, @newline]
+  def build(:pub, topic, payload, [headers: headers]) do
+    # it takes 10 bytes to add the nats header version line
+    # and 2 more for the newline between headers and payload
+    header_len = IO.iodata_length(headers) + 12
+    total_len = IO.iodata_length(payload) + header_len
+    [@hpub, " ", topic, " ", Integer.to_string(header_len), " ", Integer.to_string(total_len), "\r\nNATS/1.0\r\n", headers, @newline, payload, @newline]
```

Comment:
Not super clear so just adding a comment. Is the `headers` param already stringified? i.e. `"Header 1: A\r\nHeader 2: B\r\n"`. I'm just wondering if the output needs to be `...header, @newline, @newline...`. I see there are parse tests, but is there a build test that asserts the string form of this output?

Reply:
The `headers` param does get encoded before this (we actually encode it to `iodata` rather than a string to avoid extra allocations), but you can see this in the `def pub` and `def request` methods where we call the `prepare_headers` helper function. I did it this way so that any errors with the headers input (ie passing the wrong kinds of arguments, etc) will happen in the client process, rather than the connection process. This also minimizes the amount of time we spent on encoding in the connection process.

I don't have a test for this yet, but maybe I should add one? I ended up writing the end-to-end test which does a full round trip through a nats server and once that was working I didn't think about adding a unit test for this 🤔

Reply:
```
iex(13)> Gnat.Command.build(:pub, "yolo.topic", "intel inside, not", headers: :cow_http.headers([{"apple", "m1"}]), reply_to: "me")
[
  "HPUB",
  " ",
  "yolo.topic",
  " ",
  "me",
  " ",
  "23",
  " ",
  "40",
  "\r\nNATS/1.0\r\n",
  [["apple", ": ", "m1", "\r\n"]],
  "\r\n",
  "intel inside, not",
  "\r\n"
]
```

I just wasn't sure what `:cow_http.headers` responded with (whether the terminating newline was added or not). It is, so 👍 . Just wasn't clear from github code review.